zlib.h

```diff
@@ -109,8 +109,9 @@ typedef struct z_stream_s {
     int                   data_type;  /* best guess about the data type: binary or text
                                          for deflate, or the decoding state for inflate */
     uint32_t              adler;      /* Adler-32 or CRC-32 value of the uncompressed data */
-#if defined(ZLIB_COMPAT) && defined(X86_64)
-    uint32_t              padding;    /* pad out to the same size as the zlib struct */
+#if defined(ZLIB_COMPAT) && (defined(X86_64) || defined(ARM_64BIT_LONG))
+    uint32_t              padding;    /* pad out to the same size as the zlib struct: padding
+                                         is needed when the size of long int is 64 bits */
```

Comment:
This code is still wrong and pending removal within a week... The problem is that compiler automatically adds extra padding after variable "padding" even though the extra padding should be between "data_type" and "adler" on little-endian machines because both are 32-bit variables in zlib-ng, but "adler32" is 64-bit in zlib so still aligned differently.

Reply:
I don't know how will you deal with padding finally . What I do is just making zlib-ng work well on out platforms. So I will wait for your final modification and then look into it.

Reply:
ARM is little-endian by default, so you should have padding before and after "adler32"... both should be uint32_t when long int is 64 bits.
We will use two versions for zlib.h and the one used for ZLIB_COMPAT will be reverted to use same types as original zlib.